src/support/debug.h

```diff
@@ -44,14 +44,18 @@ void setDebugEnabled(const char* types);
 
 #else
 
+// Comment out this line if you want to build with BYN_ASSERTIONS_ENABLED=ON.
+// But be warned that this confiration is not tested.
+#error "binaryen is currently designed to be built with assertion enabled"
```

Comment:
`assertion` => `assertions`, and please also mention the `BYN_ASSERTIONS_ENABLED` in the message so users don't need to find this file

Reply:
In order to see this error a user would already have needed to turn on BYN_ASSERTIONS_ENABLED.  At that point this error will point them directly here with a line number.

It looks like this:

binaryen/src/support/debug.h:45:2: error: "binaryen is currently designed to be built with assertion enabled"

Reply:
I added second error with more context.